src/lib/parse/contracts.js

```diff
@@ -0,0 +1,261 @@
+// @flow 
+const fs = require('fs');
+const drafter = require('drafter');
+
+const endpointSorter = require('../middleware/endpoint-sorter');
+const urlParser = require('./url');
+const logger = require('../logger');
+const specSchema = require('../spec-schema');
+const http = require('./httpFetch');
+
+export type Mappings = { [string]: string }
+
+// any valid JSON-schema
+type JsonSchema = any
+
+type ResourceSchemas = {
+    request: ?JsonSchema,
+    response: ?JsonSchema
+};
+
+type Verb = string;
+type Url = string;
+
+export type Actions = {
+    [Verb]: ResourceSchemas
+};
+
+export type Resources = {
+    [Url]: Actions
+};
+
+export type Contract = {
+    fixtureFolder: string,
+    resources: Resources
+}
+export type Body = {
+    schema?: string,
+    body?: string
+};
+
+export type Example = {
+    requests: Array<Body>,
+    responses: Array<Body>
+};
+
+export type BlueprintAction = {
```

Comment:
Add comment for group of structures indicating their usage intent: test-fixture or contract.